bot/bot.py

```diff
@@ -44,9 +46,27 @@ def __init__(self, *args, redis_session: RedisSession, **kwargs):
             # Since statsd is UDP, there are no errors for sending to a down port.
             # For this reason, setting the statsd host to 127.0.0.1 for development
             # will effectively disable stats.
-            statsd_url = "127.0.0.1"
+            statsd_url = LOCALHOST
 
-        self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        self.stats = AsyncStatsClient(self.loop, LOCALHOST)
+        self._connect_statsd(statsd_url)
+
+    def _connect_statsd(self, statsd_url: str, retry_after: int = 2, attempt: int = 1) -> None:
+        """Callback used to retry a connection to statsd if it should fail."""
+        if self._statsd_timerhandle and not self._statsd_timerhandle.cancelled:
+            self._statsd_timerhandle.cancel()
+
+        if attempt >= 5:
+            log.error("Reached 5 attempts trying to reconnect AsyncStatsClient. Aborting")
+            return
+
+        try:
+            self.stats = AsyncStatsClient(self.loop, statsd_url, 8125, prefix="bot")
+        except socket.gaierror:
+            log.warning(f"Statsd client failed to connect (Attempt(s): {attempt})")
+            # Use a fallback strategy for retrying, up to 5 times.
+            self._statsd_timerhandle = self.loop.call_later(
+                retry_after, self._connect_statsd, statsd_url, retry_after * 5, attempt + 1)
```

Comment:
I think `retry_after = attempt * 2` would be better along with increasing the maximum attempts to 8. Will end up with the following wait times for each subsequent attempt:

```
1
2
4
8
16
32
64
128
``` 

This total 255 minutes or about 4.5 hours before giving up. This has a nice balance of having short wait times when the issue is brief and long wait times when it's a longer outage.